src/print/mod.rs

```diff
@@ -226,6 +229,184 @@ where
     _native_format(rows, config, w, colors, Stdout {}).await
 }
 
+fn get_printer_string<E>(
+    prn: &mut Printer<&mut String>,
+) -> Result<String, Exception<PrintError<E, Infallible>>>
+where
+    E: fmt::Debug + Error + 'static,
+{
+    prn.commit().map_err(fix_infallible)?;
+    prn.flush_buf().map_err(fix_infallible)?;
+    let mut s = String::new();
+    std::mem::swap(prn.stream, &mut s);
+    Ok(s)
+}
+
+fn is_numeric(v: &Value) -> bool {
+    matches!(
+        v,
+        Value::Int16(_)
+            | Value::Int32(_)
+            | Value::Int64(_)
+            | Value::Float32(_)
+            | Value::Float64(_)
+            | Value::BigInt(_)
+            | Value::Decimal(_)
+    )
+}
```

Comment:
This should probably get upstreamed at some point -- seems like a useful helper.

Reply:
Another option would be to have a never-failing `unwrap_infallible` method, and then do `.unwrap_exc().unwrap_infallible()`, to make it clear that we are only *really* doing an unsafe unwrapping of the `DisableFlow` case, not of any other potential errors.

That's noisier but still gets the errors out of the `to_cell` and `get_printer_string` functions.

Reply:
Though, also: who bothers error checking your writes to stdout anyway? We don't anywhere else in our codebase.